crates/bifrost/src/bifrost.rs

```diff
@@ -115,15 +115,17 @@ static_assertions::assert_impl_all!(Bifrost: Send, Sync, Clone);
 // held across an async boundary.
 pub struct BifrostInner {
     opts: Options,
+    metadata: Metadata,
```

Comment:
What is the longer term idea for providing the metadata to Bifrost? I guess with this change you want to avoid the task local field access when calling `metadata()`. Would the next step be providing the `logs` configuration as an `Updateable`?